R/utils.R

```diff
@@ -259,84 +259,333 @@ rowcol_to_dim <- function(row, col) {
-    cols <- col2int(args[[cols]])
+  rows_arg <- col2int(rows_arg, allow_null = TRUE)
+  cols_arg <- args$cols
+  x <- args$x
+  x_has_named_dims <- inherits(x, "data.frame") | inherits(x, "matrix")
 
-  } else {
+  # rows_and_cols_present <- all(c("rows", "cols") %in% nams)
 
-    if (cnam_null) col_names <- exp_name
-    if (rnam_null) row_names <- FALSE
 
-    assert_class(col_names, "logical")
-    assert_class(row_names, "logical")
+  # Find column location id if `cols` is named.
+  if (x_has_named_dims && !is.null(cols_arg)) {
+    is_cols_a_colname <- cols_arg %in% colnames(x)
 
-    # wb_dims(data.frame())
-    x <- as.data.frame(x)
-    rows <- seq_len(nrow(x) + col_names)
-    cols <- seq_len(ncol(x) + row_names)
+    if (any(is_cols_a_colname)) {
+      if (length(is_cols_a_colname) != 1) {
+        stop(
+          "Supplying multiple column names is not supported by the `wb_dims()` helper, use the `cols`  arguments instead.",
+          "\n Use a single `cols` at a time with `wb_dims()`"
+        )
+      }
+      col_name <- cols_arg
+
+      cols_arg <- which(colnames(x) == cols_arg)
+      message("Transforming col name = '", col_name, "' to `cols = ", cols_arg, "`")
+    }
+  }
 
+  if (!is.null(rows_arg)) {
+    assert_class(rows_arg, class = "integer", envir = parent.frame(n = 2), arg_nm = "rows_arg")
   }
 
-  rows <- rows + srow
-  cols <- cols + scol
+  if (!is.null(cols_arg)) {
+    cols_arg <- col2int(cols_arg)
+    assert_class(cols_arg, class = "integer", envir = parent.frame(n = 2), arg_nm = "cols_arg")
+  }
+
+  srow <- args$start_row %||% 1L
+  srow <- srow - 1L
+  scol <- col2int(args$start_col, allow_null = TRUE) %||% 1L
+  scol <- scol - 1L
+  # after this point, no assertion, assuming all elements to be acceptable
+
+
 
-  if (length(rows) == 1 && length(cols) == 1) {
+  col_names <- args$col_names
+  row_names <- args$row_names
+
+  rows_arg
+  cols_arg
+
+  if (!all(length(scol) == 1, length(srow) == 1, scol >= 0, srow >= 0)) {
+    stop("Internal error. At this point scol and srow should have length 1.")
+  }
+
+
+  # if `!x` return early
+  if (!x_present) {
+    row_span <- srow + rows_arg %||% 1L
+    col_span <- scol + cols_arg %||% 1L
+    if (length(row_span) == 1 && length(col_span) == 1) {
+      # A1
+      row_start <- row_span
+      col_start <- col_span
+      dims <- rowcol_to_dim(row_start, col_start)
+    } else {
+      # A1:B2
+
+      dims <- rowcol_to_dims(row_span, col_span)
+    }
+    return(dims)
+  }
+  # Making sure that at this point, we only cover the case for `x`
+
+  col_names <- col_names %||% x_has_named_dims
+  row_names <- row_names %||% FALSE
+  if (!col_names && row_names && x_has_named_dims) {
+    warning("The combination of `row_names = TRUE` and `col_names = FALSE` is not recommended.",
+      "`col_names` allows to select the region that contains the data only.",
+      "`row_names` = TRUE adds row numbers if the data doesn't have rownames.",
+      call. = FALSE
```

Comment:
Interesting take, but now that I think about, maybe this should be possible. I frequently create data frames with two header columns and write the data straight underneath. The latter, without column names. In such a case it might be ...
But having written this, I don't think I have a use case, where I need to add rownames without another option. Note to self: what does `call.` do?

Reply:
Would have to think about this

call. prevents printing 

Warning in wb_dims(.....)

No need for it when you named the function in the meaaage (redundancy)

Reply:
I removed this warning... But reviewed the API, so that col_names is handled a bit differently